src/presentation/http/router/noteSettings.test.ts

```diff
@@ -86,15 +125,43 @@ describe('NoteSettings API', () => {
     });
 
     test('Returns 403 when the note is not public, the user is not authorized', async () => {
-      const notPublicNote = notes.find(newNote => {
-        const settings = noteSettings.find(ns => ns.note_id === newNote.id);
+      /**
+       * truncate all tables, which are needed
+       * restart autoincrement sequences for data to start with id 1
+       *
+       * TODO get rid of restarting database data in tests (move to beforeEach)
+       */
+      await global.db.truncateTables();
+
+      /** create test user */
+      const user = await global.db.insertUser({
+        email: 'a@a.com',
+        name: 'Test user 1',
+      });
+
+      /** create test note for created user */
+      const note = await global.db.insertNote({
+        creatorId: user.userId,
+        publicId: 'Pq1T9vc23Q',
+      });
 
-        return settings!.is_public === false;
+      /** create test note settings for created note */
+      await global.db.insertNoteSetting({
+        noteId: note.noteId,
+        isPublic: false,
+        invitationHash: 'Hzh2hy4igf',
```

Comment:
does `invitationHash` it matter for this test case? Can we remove unused fields for simplicity?

Reply:
i think this is compulsory parameter, we can make it optional and create default hash in the method if i get your idea right

Reply:
yes, it would be better